src/jit/decomposelongs.cpp

```diff
@@ -796,11 +799,9 @@ GenTree* DecomposeLongs::DecomposeInd(LIR::Use& use)
         addrBase->TypeGet(), addrBase->AsLclVarCommon()->GetLclNum(), BAD_IL_OFFSET);
     GenTreePtr addrHigh = new(m_compiler, GT_LEA) GenTreeAddrMode(TYP_REF, addrBaseHigh, nullptr, 0, genTypeSize(TYP_INT));
     GenTreePtr indHigh = new (m_compiler, GT_IND) GenTreeIndir(GT_IND, TYP_INT, addrHigh, nullptr);
-    
+
     // Insert the nodes into the block
-    m_blockRange.InsertAfter(addrBaseHigh, indLow);
-    m_blockRange.InsertAfter(addrHigh, addrBaseHigh);
-    m_blockRange.InsertAfter(indHigh, addrHigh);
+    m_blockRange.InsertAfter(LIR::SeqTree(m_compiler, indHigh), indLow);
```

Comment:
Calling `SeqTree` here is probably a bit less efficient than just manually inserting the nodes and adds some debt for later changes that I'd like to make s.t. we are able to do shallow cost calculation on LIR nodes. Did you switch to using this because it initializes cost information or for the sake of brevity? If it is the former, I'd prefer that we just call `gtPrepareCost` and leave the manual insertion as-is. Same goes for the similar changes in `DecomposeStoreInd`.